unifi_console/datadog_checks/unifi_console/check.py

```diff
@@ -0,0 +1,111 @@
+from typing import cast
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.unifi_console.api import APIConnectionError, UnifiAPI
+from datadog_checks.unifi_console.config import UnifiConfig
+from datadog_checks.unifi_console.mertrics import Metric
+from datadog_checks.unifi_console.types import Check, ControllerInfo
+
+
+class UnifiConsoleCheck(AgentCheck):
+
+    # This will be the prefix of every metric and service check the integration sends
+    __NAMESPACE__ = "unifi_console"
+
+    def __init__(self, name, init_config, instances):
+        instances[0]['persist_connections'] = instances[0].get('persist_connections', True)
+        super(UnifiConsoleCheck, self).__init__(name, init_config, instances)
+
+        # Use self.instance to read the check configuration
+        self._config = UnifiConfig(self.instance, self.init_config, self.log)
```

Comment:
As @hithwen mentioned, I think we can just load the config here and can bypass the need to instantiate a `UnifiConfig` object just to read config options. Example:
https://github.com/DataDog/integrations-core/blob/master/couchbase/datadog_checks/couchbase/couchbase.py#L50-L55

Reply:
What would be the benefit of it? I would need to rewrite lots of functions and tests.